### PR TITLE
Deprecate `--v2-deprecation` and schedule to remove it in 3.8

### PR DESCRIPTION
**Pull request by ahrtr**
Resolve https://github.com/etcd-io/etcd/issues/18994

cc @fuweid @serathius @ivanvc @jmhbnz

### CODE REVIEW

**Review by fuweid — APPROVED** [review]

**Comment by ahrtr** [comment]
ping @serathius

File: server/config/v2_deprecation.go [hunk 1]
```diff
@@ -17,47 +17,50 @@ package config
...
-	// Deprecated: Please use V2Depr2Gone
+
+	// Deprecated: to be decommissioned in 3.7. Please use V2Depr2Gone.
+	// TODO: remove in 3.7
 	//revive:disable-next-line:var-naming
 	V2_DEPR_2_GONE = V2Depr2Gone
 
-	// Default deprecation level.
+	// V2DeprDefault is the default deprecation level.
 	V2DeprDefault = V2Depr1WriteOnly
```
Comment (serathius) [inline server/config/v2_deprecation.go:60]: Is `V2Depr1WriteOnly` the correct v2 default config for v3.6? From above 
```
	// V2Depr1WriteOnly means only writing v2store is allowed.
	// Default in v3.6.  Meaningful v2 state is not allowed.
	// The V2 files are maintained for v3.5 rollback.
```

However we plan to panic on any v2 state. Should we reconcile this?

Comment (ahrtr) [reply server/config/v2_deprecation.go:60]: > Is `V2Depr1WriteOnly` the correct v2 default config for v3.6?

YES.



> However we plan to panic on any v2 state.

In 3.6, we only allow membership data in v2state. We will keep it as it's in 3.6. In 3.7, the default level will be `V2Depr1WriteOnlyDrop`, and we will remove any v2 snapshot files if found.

https://github.com/etcd-io/etcd/blob/fbac2a064c381d6f8c74c4d9d01fde816e200eef/server/etcdserver/bootstrap.go#L407-L409

Comment (serathius) [reply server/config/v2_deprecation.go:60]: I thought `V2DeprecationEnum` was about key-value data stored in v2 store, not about membership data. 

To clarify, enums are not accurate to the current state, as the plan has changed in https://github.com/etcd-io/etcd/issues/12913#issuecomment-1531941532.

**Review by serathius — APPROVED** [review]

**Comment by ahrtr** [comment]
Thanks both @fuweid and @serathius !

**Pull request merged**